packages/cli/cloud/src/services/strapi-info-save.ts

```diff
@@ -1,22 +1,36 @@
 import fse from 'fs-extra';
 import path from 'path';
-import type { ProjectInfos } from './cli-api';
+import type { ProjectInfo } from './cli-api';
 
 export const LOCAL_SAVE_FILENAME = '.strapi-cloud.json';
 
 export type LocalSave = {
-  project?: Omit<ProjectInfos, 'id'>;
+  project?: Omit<ProjectInfo, 'id'>;
 };
 
-export async function save(data: LocalSave, { directoryPath }: { directoryPath?: string } = {}) {
+export async function addEnvironment(
+  targetEnvironment: string,
+  { directoryPath }: { directoryPath?: string } = {}
+) {
   const alreadyInFileData = await retrieve({ directoryPath });
```

Comment:
What if "alreadyInFileData" is undefined or empty ({})? I think that's an error case that we need to add, otherwise we will just have the "targetEnvironment" in the save file (= incomplete / unexpected data).

Reply:
It shouldn't be empty cause this function wouldn't be called otherwise. However, a check should be done in case the reading operation fails. Good catch!